docs/markdown/Reference-manual.md

```diff
@@ -1683,6 +1683,12 @@ the following methods:
 - `get_id()` returns a string identifying the compiler. For example,
   `gcc`, `msvc`, [and more](Reference-tables.html#compiler-ids).
 
+- `get_argument_type()` *(new in 0.49.0) returns a string identifying the type
+  of arguments the compiler takes. Can be one of `gcc`, `msvc`, or `other`.
+  This method is useful for identifying compilers that are not gcc or msvc,
+  but accept the same kinds of arguments as one of those two compilers such as
+  clang or icc.
```

Comment:
Can we just leave it undefined what it says in other cases, rather than explicitly saying it must say "other"?

I wonder if we should say something like "if you need to use this outside of the meson test suite, that probably indicates a missing feature in meson" here?

Reply:
I'm pretty certain for as we add more GCC and MSVC like compilers people are going to want this in their build system, since it behaves like GCC if the build machine is MacOS or Linux, or like MSVC if the build machine is Windows.
```meson
cc = meson.get_compiler('c')
if ['clang-cl', 'msvc', 'wacom'].contains(cc.get_id()) or (cc.get_id() == 'intel' and build_machine.system() == 'windows')
```
or:
```meson
cc = meson.get_compiler('c')
if cc.get_argument_type() == 'msvc'
```

Reply:
You've omitted the body of the `if`, which presumably contains something that meson lacks the means to express more portably.  So I think we are in agreement here :)